File: sample/src/main/java/contacts/sample/cheatsheet/groups/java/DeleteGroupsActivity.java
package contacts.sample.cheatsheet.groups.java;

import static contacts.core.WhereKt.equalTo;

import android.app.Activity;

import java.util.List;

import contacts.core.ContactsFactory;
import contacts.core.GroupsFields;
import contacts.core.entities.Group;
import contacts.core.groups.GroupsDelete;

public class DeleteGroupsActivity extends Activity {

    GroupsDelete.Result deleteGroups(List<Group> groups) {
        return ContactsFactory.create(this).groups().delete().groups(groups).commit();
    }

    GroupsDelete.Result deleteGroupWithId(long groupId) {
        return ContactsFactory.create(this).groups().delete().groupsWithId(groupId).commit();
    }

    GroupsDelete.Result deleteUserCreatedGroupFromAllGoogleAccounts() {
        return ContactsFactory.create(this)
                .groups()
                .delete()
                .groupsWhere(equalTo(GroupsFields.AccountType, "com.google"))
                .commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/groups/kotlin/DeleteGroupsActivity.kt
package contacts.sample.cheatsheet.groups.kotlin

import android.app.Activity
import contacts.core.*
import contacts.core.entities.Group
import contacts.core.groups.GroupsDelete

class DeleteGroupsActivity : Activity() {

    fun deleteGroups(groups: List<Group>): GroupsDelete.Result =
        Contacts(this).groups().delete().groups(groups).commit()

    fun deleteGroupWithId(groupId: Long): GroupsDelete.Result =
        Contacts(this).groups().delete().groupsWithId(groupId).commit()

    fun deleteUserCreatedGroupFromAllGoogleAccounts(): GroupsDelete.Result = Contacts(this)
        .groups()
        .delete()
        .groupsWhere { AccountType equalTo "com.google" }
        .commit()
}